drivers/modem/modem_cellular.c

```diff
@@ -0,0 +1,1226 @@
+}
+
+static void modem_cellular_open_dlci2_event_handler(struct modem_cellular_data *data,
+						    enum modem_cellular_event evt)
+{
+	switch (evt) {
+	case MODEM_CELLULAR_EVENT_DLCI2_OPENED:
+		modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_RUN_DIAL_SCRIPT);
+
+		break;
+
+	default:
+		break;
+	}
+}
+
+static int modem_cellular_on_open_dlci2_state_leave(struct modem_cellular_data *data)
+{
+	modem_pipe_release(data->dlci2_pipe);
+
+	return 0;
+}
+
+static int modem_cellular_on_run_dial_script_state_enter(struct modem_cellular_data *data)
+{
+	if (modem_chat_attach(&data->chat, data->dlci2_pipe) < 0) {
+		return -EAGAIN;
+	}
+
+	modem_cellular_start_timer(data, K_MSEC(500));
+
+	return 0;
+}
+
+static void modem_cellular_run_dial_script_event_handler(struct modem_cellular_data *data,
+							 enum modem_cellular_event evt)
+{
+	switch (evt) {
+	case MODEM_CELLULAR_EVENT_TIMEOUT:
+		modem_chat_script_run(&data->chat, &connect_chat_script);
+
+		break;
+
+	case MODEM_CELLULAR_EVENT_SCRIPT_SUCCESS:
+		modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_REGISTER);
+
+		break;
+
+	default:
+		break;
+	}
+}
+
+static int modem_cellular_on_run_dial_script_state_leave(struct modem_cellular_data *data)
+{
+	modem_chat_release(&data->chat);
+
+	return modem_ppp_attach(data->ppp, data->dlci2_pipe);
+}
+
+static int modem_cellular_on_register_state_enter(struct modem_cellular_data *data)
+{
+	if (modem_chat_attach(&data->chat, data->dlci1_pipe) < 0) {
+		return -EAGAIN;
+	}
+
+	modem_cellular_start_timer(data, K_SECONDS(2));
+
+	return modem_chat_script_run(&data->chat, &net_stat_chat_script);
+}
+
+static bool modem_cellular_is_registered(struct modem_cellular_data *data)
+{
+	return (data->registration_status == 5) && (data->packet_service_attached == 1);
+}
+
+static void modem_cellular_register_event_handler(struct modem_cellular_data *data,
+						  enum modem_cellular_event evt)
+{
+	const struct modem_cellular_config *config =
+		(const struct modem_cellular_config *)data->dev->config;
+
+	switch (evt) {
+	case MODEM_CELLULAR_EVENT_SUSPEND:
+		if (modem_cellular_gpio_is_enabled(&config->power_gpio)) {
+			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_POWER_OFF);
+		} else {
+			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_IDLE);
+		}
+
+		break;
+
+	case MODEM_CELLULAR_EVENT_SCRIPT_SUCCESS:
+		if (modem_cellular_is_registered(data)) {
+			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_ROAMING);
+		}
+
+		break;
+
+	case MODEM_CELLULAR_EVENT_TIMEOUT:
+		modem_cellular_start_timer(data, K_SECONDS(2));
+
+		modem_chat_script_run(&data->chat, &net_stat_chat_script);
+
+		break;
+
+	default:
+		break;
+	}
+}
+
+static int modem_cellular_on_register_state_leave(struct modem_cellular_data *data)
+{
+	modem_cellular_stop_timer(data);
+
+	modem_chat_release(&data->chat);
+
+	return 0;
+}
+
+static int modem_cellular_on_roaming_state_enter(struct modem_cellular_data *data)
+{
+	net_if_up(modem_ppp_get_iface(data->ppp));
```

Comment:
Ditto -> `net_if_dormant_off()`?

Reply:
I will give it a shot :)

Reply:
Updated as part of PPP L2 interface update